boards/arm/nucleo_f207zg/pinmux.c

```diff
@@ -22,7 +22,7 @@ static const struct pin_config pinconf[] = {
 	{STM32_PIN_PG14, STM32F2_PINMUX_FUNC_PG14_USART6_TX},
 	{STM32_PIN_PG9, STM32F2_PINMUX_FUNC_PG9_USART6_RX},
 #endif
-#ifdef CONFIG_ETH_STM32_HAL
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(mac), okay) && CONFIG_NET_L2_ETHERNET
```

Comment:
It would be better to have `#if DT_NODE_HAS_STATUS(DT_NODELABEL(mac), okay) && defined(CONFIG_NET_L2_ETHERNET)`, could you change these to have the defined() check.

Reply:
@jukkar I'd be happy to change, but there's a whole bunch of similar lines across `boards/*/*/pinmux.c` that should need to be aligned if this is really required, and hence, it might be wiser raising a dedicated issue to fix the whole tree.